components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -298,9 +304,12 @@ func (g *RTKMovementSensor) Connect(casterAddr, user, pwd string, maxAttempts in
 		g.logger.Errorf("Can't connect to NTRIP caster: %s", err)
 		return err
 	}
-	g.ntripClient.Client = c
+
 	g.logger.Debug("Connected to NTRIP caster")
+	g.mu.Lock()
+	defer g.mu.Unlock()
 
+	g.ntripClient.Client = c
```

Comment:
Should this function and `GetStream` actually be defined on `NtripInfo`? That way, the Ntrip client wrapper can have its own mutex.

Reply:
apologies, I missed the latter half of your original PR comment. I do think at least this is worth it though, happy to discuss

Reply:
Yeah, we're on the same page here: worth thinking about more, but not something that could get done on just the bugfix day, and low enough priority that we've got other stuff to work on for the moment.